components/unit/file-explorer/viewer/UnitFileExplorerViewerCsv.vue

```diff
@@ -28,11 +28,24 @@ export default {
         await this.getContentFromFilename(filename)
       },
       immediate: true
+    },
+    loading: {
+      handler(value) {
+        this.$emit('loading', value)
+      },
+      immediate: true
     }
   },
   methods: {
     async getContentFromFilename(filename) {
-      await this.setLoading(true)
+      this.loading = true
+
+      // small timeout to allow the navigation drawer transition to finish
+      // note: nextTick() will not work for that purpose
+      // because it only waits for the virtual DOM to update and not the page
+      // https://stackoverflow.com/a/47636157/8238129
+      await setTimeoutPromise(200)
```

Comment:
This line can be removed and the hourglass will still appear. However, we need a timeout for the navigation drawer animation to finish.